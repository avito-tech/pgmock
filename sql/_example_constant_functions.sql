begin;

select pgmock.mock($$
    {
        "constant_functions": [
            { "name": "DEFAULT_CAT_NAME", "value": "Snezhok" },
            { "name": "DEFAULT_CAT_AGE", "value": 6, "returns": "smallint" },
            { "name": "DEFAULT_MICE_CAUGHT", "value": 128 }
        ]
    }
$$);

select pg_temp.DEFAULT_CAT_NAME();

select pg_temp.DEFAULT_CAT_AGE();

select pg_temp.DEFAULT_MICE_CAUGHT();

rollback;