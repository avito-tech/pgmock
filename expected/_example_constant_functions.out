begin;
select pgmock.mock($$
    {
        "constant_functions": [
            { "name": "DEFAULT_CAT_NAME", "value": "Snezhok" },
            { "name": "DEFAULT_CAT_AGE", "value": 6, "returns": "smallint" },
            { "name": "DEFAULT_MICE_CAUGHT", "value": 128 }
        ]
    }
$$);
 mock 
------
 
(1 row)

select pg_temp.DEFAULT_CAT_NAME();
 default_cat_name 
------------------
 Snezhok
(1 row)

select pg_temp.DEFAULT_CAT_AGE();
 default_cat_age 
-----------------
               6
(1 row)

select pg_temp.DEFAULT_MICE_CAUGHT();
 default_mice_caught 
---------------------
                 128
(1 row)

rollback;
